fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/MessageTuple.java

```diff
@@ -52,10 +48,6 @@ public Object getObject(int position) throws InvalidColumnReferenceException {
             final var field = message.getField(message.getDescriptorForType().getFields().get(position));
             if (fieldDescriptor.getType() == Descriptors.FieldDescriptor.Type.ENUM) {
                 return ((Descriptors.EnumValueDescriptor) field).getName();
-            } else if (fieldDescriptor.getType() == Descriptors.FieldDescriptor.Type.MESSAGE && fieldDescriptor.getMessageType().equals(TupleFieldsProto.UUID.getDescriptor())) {
```

Comment:
Is this going to be a problem when we go to add UUID support back? Because this is based on the the type of the `fieldDescriptor` and not the type system's `Type`, it seems like for any query with a UUID, the thing that will determine whether it gets treated as a UUID or a struct will be the library version, rather than the plan. Which, is a little weird, and not really consistent with the plan serialization stability guarantees that we're trying to provide.

It may be workable though, and the alternative would be to thread through the `Type` information into the row implementation, which may be a bit more work than we want to do here.

Reply:
Fixed this as stated above.